packages/headless/src/controllers/core/facets/category-facet/headless-core-category-facet.ts

```diff
@@ -0,0 +1,358 @@
+  sortCriteria: CategoryFacetSortCriterion;
+
+  /** Returns `true` if a search is in progress, and `false` if not. */
+  isLoading: boolean;
+
+  /** Returns `true` if there's at least one non-idle value, and `false` if not. */
+  hasActiveValues: boolean;
+
+  /** Returns `true` if there are more values to display and, `false` if not. */
+  canShowMoreValues: boolean;
+
+  /** Returns `true` if fewer values can be displayed, and `false` if not. */
+  canShowLessValues: boolean;
+
+  /** The state of the facet's searchbox. */
+  facetSearch: CategoryFacetSearchState;
+}
+
+export interface CategoryFacetSearch {
+  /**
+   * Updates the facet search query.
+   *
+   * @param text - The facet search query.
+   * */
+  updateText(text: string): void;
+
+  /**
+   * Shows more facet search results.
+   * */
+  showMoreResults(): void;
+
+  /**
+   * Performs a facet search.
+   * */
+  search(): void;
+
+  /**
+   * Selects a facet search result.
+   *
+   * @param value - The search result to select.
+   * */
+  select(value: CategoryFacetSearchResult): void;
+
+  /**
+   * Resets the query and empties the values.
+   * */
+  clear(): void;
+
+  /**
+   * Updates the facet value captions.
+   * @param captions - A dictionary that maps index field values to facet value display names.
+   */
+  updateCaptions(captions: Record<string, string>): void;
+}
+
+export interface CategoryFacetSearchState {
+  /**
+   * The facet search results.
+   * */
+  values: CategoryFacetSearchResult[];
+
+  /**
+   * Returns `true` if the facet search is in progress, and `false` if not.
+   * */
+  isLoading: boolean;
+
+  /**
+   * Whether more values are available.
+   * */
+  moreValuesAvailable: boolean;
+
+  /**
+   * The current query in the facet search box.
+   */
+  query: string;
+}
+
+export interface CategoryFacetSearchResult {
+  /**
+   * The custom facet value display name, as specified in the `captions` argument of the facet request.
+   */
+  displayValue: string;
+
+  /**
+   * The original facet value, as retrieved from the field in the index.
+   */
+  rawValue: string;
+
+  /**
+   * An estimate of the number of result items that match both the current query and
+   * the filter expression which would be generated if the facet value were selected.
+   */
+  count: number;
+
+  /**
+   * The hierarchical path to the facet value.
+   */
+  path: string[];
+}
+
+/**
+ * Creates a `CategoryFacet` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `CategoryFacet` properties.
+ * @returns A `CategoryFacet` controller instance.
+ * */
+export function buildCoreCategoryFacet(
+  engine: CoreEngine,
+  props: CategoryFacetProps
+): CategoryFacet {
+  if (!loadCategoryFacetReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const facetId = determineFacetId(engine, props.options);
```

Comment:
There were a couple of places that were using `options.facetId` directly instead of this, and I am hypothetising that it was causing issues in the analytics if you didn't pass a `facetId` in the options.

So this core controller now uses `facetId` all across the board.